programs/disks/disks.cpp

```diff
@@ -0,0 +1,354 @@
+#include <stdlib.h>
+#include <fcntl.h>
+#include <map>
+#include <iostream>
+#include <iomanip>
+#include <optional>
+#include <filesystem>
+#include <string>
+
+#include <boost/program_options.hpp>
+
+#include <base/argsToConfig.h>
+
+#include <Common/TerminalSize.h>
+
+#include <Columns/ColumnString.h>
+
+#include <Formats/registerFormats.h>
+
+#include <Poco/Util/HelpFormatter.h>
+
+#include <Disks/registerDisks.h>
+#include <Disks/IDisk.h>
+
+#include "disks.h"
+
+namespace fs = std::filesystem;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+std::string DisksFunc::getDefaultConfigFileName() const
+{
+    return "/etc/clickhouse-server/config.xml";
+}
+
+void DisksFunc::listDisks()
+{
+    while(argumentsToDo[0] != "list-disks")
+        argumentsToDo.erase(argumentsToDo.begin());
```

Comment:
This code duplicates same for **ls** and would be for all other commands, it can be easily generalized in code before actual command call